### PR TITLE
net: ethernet: Define and use Ethernet frame and datagram size

### PR DESCRIPTION
**Pull request by finikorg**
Remove magic numbers from Ethernet drivers and tests by defining
NET_ETH_MAX_DATAGRAM_SIZE and NET_ETH_MAX_FRAME_SIZE.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]

**Review by ozhuraki — APPROVED** [review]
LGTM

A few optional notes/further refactorings:

-	It might be perhaps clearer to retain the name as *_MTU (Maximum Transmission Unit). Actually _MAX_DATAGRAM  is an MTU.
-	And perhaps to think if one in this pair: NET_ETH_MAX_DATAGRAM_SIZE/NET_ETH_MAX_FRAME_SIZE can be eliminated.

**Review by pfalcon — APPROVED** [review]

File: include/net/ethernet.h [hunk 1]
```diff
@@ -72,6 +72,9 @@ struct net_eth_addr {
 #endif
 
 #define NET_ETH_MINIMAL_FRAME_SIZE	60
+#define NET_ETH_MAX_DATAGRAM_SIZE	1500
```
Comment (tbursztyka) [inline include/net/ethernet.h]: I would be also for NET_ETH_MTU on that one, it is semantically more relevant

Comment (jukkar) [reply include/net/ethernet.h]: Yes, calling it MTU looks actually a bit better here.

**Review by jukkar — CHANGES_REQUESTED** [review]
Lets rework this a bit more, so s/DATAGRAM/MTU/g

File: include/net/ethernet.h (hunk 1 above)
Comment (finikorg) [reply include/net/ethernet.h]: Changed with filter-edit, hopefully nothing breaks

Comment (finikorg) [reply include/net/ethernet.h]: BTW, interesting to look at Linux notion: https://github.com/torvalds/linux/blob/master/include/uapi/linux/if_ether.h#L36

**Review by jukkar — APPROVED** [review]

**Pull request merged**